presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java

```diff
@@ -188,29 +194,34 @@ private boolean isAggregationPushdownSupported(AggregationNode partialAggregatio
             return true;
         }
 
-        private boolean isSupportedOrcTypeForMinMax(Type type)
+        private boolean isNotSupportedOrcTypeForMinMax(Type type)
         {
-            return !BOOLEAN.equals(type) &&
-                    type.getJavaType() != boolean.class &&
-                    !isRowType(type) &&
-                    !isArrayType(type) &&
-                    !isMapType(type) &&
-                    !TINYINT.equals(type) &&
-                    !VARBINARY.equals(type) &&
-                    !TIMESTAMP.equals(type) &&
-                    ((!VARBINARY.equals(type) && !VARCHAR.equals(type)) ||
-                            isPartialAggregationPushdownForVariableLengthDatatypesEnabled(session));
+            return BOOLEAN.equals(type) ||
+                    type.getJavaType() == boolean.class ||
+                    isRowType(type) ||
+                    isArrayType(type) ||
+                    isMapType(type) ||
+                    TINYINT.equals(type) ||
+                    VARBINARY.equals(type) ||
+                    TIMESTAMP.equals(type) ||
+                    isNotSupportedForVariableLengthDataType(type);
         }
 
-        private boolean isSupportedParquetTypeForMinMax(Type type)
+        private boolean isNotSupportedParquetTypeForMinMax(Type type)
         {
-            return !BOOLEAN.equals(type) &&
-                    type.getJavaType() != boolean.class &&
-                    !isRowType(type) &&
-                    !isArrayType(type) &&
-                    !isMapType(type) &&
-                    ((!VARBINARY.equals(type) && !VARCHAR.equals(type)) ||
-                            isPartialAggregationPushdownForVariableLengthDatatypesEnabled(session));
+            return BOOLEAN.equals(type) ||
+                    type.getJavaType() == boolean.class ||
+                    isRowType(type) ||
+                    isArrayType(type) ||
+                    isMapType(type) ||
+                    isNotSupportedForVariableLengthDataType(type);
+        }
+
+        private boolean isNotSupportedForVariableLengthDataType(Type type)
+        {
+            boolean isVariableLengthType = VARBINARY.equals(type) || VARCHAR.equals(type);
+            boolean isPushdownNotEnabled = !isPartialAggregationPushdownForVariableLengthDatatypesEnabled(session);
```

Comment:
Is it a cheap check? If it's expensive consider executing only for the required varbinary/varchar types.

Reply:
Looks to be a map look up . Added short circuiting to skip executing if the first check fails.